src/sentry/tagstore/snuba/backend.py

```diff
@@ -706,18 +718,10 @@ def get_tag_value_paginator_for_projects(
             if converted_query is not None:
                 conditions.append([snuba_key, ">=", converted_query - FUZZY_NUMERIC_DISTANCE])
                 conditions.append([snuba_key, "<=", converted_query + FUZZY_NUMERIC_DISTANCE])
-        elif key == PROJECT_ALIAS:
-            project_filters = {
-                "id__in": projects,
-            }
-            if query:
-                project_filters["slug__icontains"] = query
-            project_queryset = Project.objects.filter(**project_filters).values("id", "slug")
-            project_slugs = {project["id"]: project["slug"] for project in project_queryset}
-            if project_queryset.exists():
-                projects = [project["id"] for project in project_queryset]
-                snuba_key = "project_id"
-                dataset = Dataset.Discover
+        elif has_matching_project_slugs:
```

Comment:
Think you're going to get a referenced before assignment for other tags here

Reply:
fixed